lsp/jsonnet_ls.lua

```diff
@@ -0,0 +1,37 @@
+--- @brief
+---
+--- https://github.com/grafana/jsonnet-language-server
+---
+--- A Language Server Protocol (LSP) server for Jsonnet.
+---
+--- The language server can be installed with `go`:
+--- ```sh
+--- go install github.com/grafana/jsonnet-language-server@latest
+--- ```
+local util = require 'lspconfig.util'
+
+-- common jsonnet library paths
+local function jsonnet_path(root_dir)
+  local paths = {
+    root_dir .. '/lib',
+    root_dir .. '/vendor',
+  }
+  return table.concat(paths, ':')
+end
+
+local root_dir = function(fname)
+  return util.root_pattern 'jsonnetfile.json'(fname)
+    or vim.fs.dirname(vim.fs.find('.git', { path = fname, upward = true })[1])
+end
+
+return {
+  cmd = { 'jsonnet-language-server' },
+  cmd_env = {
+    JSONNET_PATH = vim.env.JSONNET_PATH or jsonnet_path(root_dir()),
+  },
+  filetypes = {
+    'jsonnet',
+    'libsonnet',
+  },
+  root_markers = { '.git' },
```

Comment:
Shouldn't this include `jsonnetfile.json`? Note that the order of these are important.